modules/ting_covers_addi/lib/addi-client/AdditionalInformationService.php

```diff
@@ -217,10 +221,29 @@ protected function extractAdditionalInformation($id_name, $response) {
               // now.
           }
         }
+      }
 
-        $additional_info = new AdditionalInformation($thumbnail_url, $detail_url);
-        $additional_informations[$info->identifier->$id_name] = $additional_info;
+      // In some cases moreinfo will return external URLs to a covers supplied
+      // by the owner of the record.
+      $extern_url = NULL;
+      if (isset($info->externUrl)) {
+        // Sometimes moreinfo will provide several links. Since it's external we
+        // have no general way of detecting the quality/size of each image link
+        // and therefore we will just have to pick the first one.
```

Comment:
It didn't handle it pretty well. There was broken image tags in the frontend and internal server error because Drupal tries to use image effects on files that are not images. I therefore updated the PR to verify that the downloaded file is an image.

Reply:
Thanks for investigating this.

You mentioned yourself that using `getimagesize` could be a performance issue because it requires that files are retrieved before they can be analyzed. That could be an issue although I am not sure above the extent of it.

I do not have any experience with the external url data provided by the service (how often it is populated, how many entries are typically provided, how often are urls invalid) but if we want to guard against the case where multiple entries are returned and the first one(s) are invalid we could curl each url, extract the content type of the response (that is in the header so we do not have to download anything) and exclude any urls with non-image content types.

Reply:
I'll see if I can find some time to assess the extend of the problem and if it's worth using anymore time on.

During my tests I only found a few cases where the would be more than URL. If I remember correctly all from Ereolen Global and Comics Plus seemed to always only have one URL, and it was only some from Pressreader that had multiple.
I didn't find any invalid URLs/Broken files, but with the amount of records that can potentially have an external URL (over 20.000 from just the sources I've looked at), this will surely happen some time and at least we handle that situation gracefully now.